builtin/mainmenu/tab_server.lua

```diff
@@ -67,31 +67,20 @@ local function main_button_handler(this, fields, name, tabdata)
 
 	if fields["srv_worlds"] ~= nil then
 		local event = core.explode_textlist_event(fields["srv_worlds"])
-
 		local selected = core.get_textlist_index("srv_worlds")
-		if selected ~= nil then
-			local filename = menudata.worldlist:get_list()[selected].path
-			local worldconfig = modmgr.get_worldconfig(filename)
-			filename = filename .. DIR_DELIM .. "world.mt"
-
-			if worldconfig.creative_mode ~= nil then
-				core.setting_set("creative_mode", worldconfig.creative_mode)
-			else
-				local worldfile = Settings(filename)
-				worldfile:set("creative_mode", core.setting_get("creative_mode"))
-				if not worldfile:write() then
-					core.log("error", "Failed to write world config file")
-				end
-			end
-			if worldconfig.enable_damage ~= nil then
-				core.setting_set("enable_damage", worldconfig.enable_damage)
-			else
-				local worldfile = Settings(filename)
-				worldfile:set("enable_damage", core.setting_get("enable_damage"))
-				if not worldfile:write() then
-					core.log("error", "Failed to write world config file")
-				end
-			end
+		local creative_mode = menu_get_worldmt(selected, "creative_mode")
+
+		if creative_mode ~= nil then
```

Comment:
a bit inconsistant, if you don't have a newline on line 79 / 80, maybe don't have one here. :P